x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,66 @@
+package keeper
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
+)
+
+// Simple model, donate tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DistributeToProviderValidatorSet(ctx sdk.Context) error {
+	if !k.shouldTransmit(ctx) {
+		return nil
+	}
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	tokens := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+	for _, token := range tokens {
+
+		if !strings.HasPrefix(token.Denom, "ibc/") {
+			denomTrace := ibctypes.ParseDenomTrace(token.Denom)
+			token.Denom = denomTrace.IBCDenom()
+		}
```

Comment:
Ahh sorry, I this this block should actually just be removed. Also this conditional should be `if strings.HasPrefix...` anyway

We already reconstruct the denom within `SendTransfer`: 

https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/relay.go#L97
https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/relay.go#L407

Reply:
We should also have a test for this

Reply:
Alright remove the block, I'm going to leave this "unresolved" as a reminder about adding a test for other ibc denoms.